projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py

```diff
@@ -379,14 +379,22 @@ def check_job_execution_finished(self) -> Optional[str]:
         )
         if not execution_list:
             return None
-        latest_end_date = max(
-            datetime.fromisoformat(e["end_time"]) for e in execution_list
-        )
-        return [
-            e["status"]
-            for e in execution_list
-            if datetime.fromisoformat(e["end_time"]) == latest_end_date
-        ].pop()
+
+        try:
+            latest_end_date = max(
+                datetime.fromisoformat(str(e["end_time"])) for e in execution_list
+            )
+            return [
+                e["status"]
+                for e in execution_list
+                if datetime.fromisoformat(str(e["end_time"])) == latest_end_date
+            ].pop()
+        except Exception as e:
```

Comment:
What makes you believe the exception is not propagated?
The only errors that are swallowed, are ValueErrors, as they tend to cause false negatives (the heartbeat test fails, even if all the steps are successful, just because the value of `end_time` is not yet available at the time of the check)

Reply:
Ok. I guess what I mean is do we need the `except Exception as e:` and the log.info ..? The exception would be propagated to the caller and logs eventually at appropriate place ?

Reply:
Fair point. Removed the general exception handling and refactored the code, so that the exception handling deals only with issues with datetime format. Thanks for the review